Content.Server/GameTicking/GameTicker.Lobby.cs

```diff
@@ -52,23 +52,22 @@ private string GetInfoText()
             var playerCount = $"{_playerManager.PlayerCount}";
             var readyCount = _playerGameStatuses.Values.Count(x => x == PlayerGameStatus.ReadyToPlay);
 
-            StringBuilder stationNames = new StringBuilder();
-            if (_stationSystem.Stations.Count != 0)
+            var stationNames = new StringBuilder();
+            var query =
+                EntityQueryEnumerator<StationJobsComponent, StationSpawningComponent, MetaDataComponent>();
+
+            var foundOne = false;
+
+            while (query.MoveNext(out _, out _, out var meta))
             {
-                foreach (EntityUid entUID in _stationSystem.Stations)
-                {
-                    StationDataComponent? stationData = null;
-                    MetaDataComponent? metaData = null;
-                    if (Resolve(entUID, ref stationData, ref metaData, logMissing: true))
-                    {
-                        if (stationNames.Length > 0)
-                            stationNames.Append('\n');
-
-                        stationNames.Append(metaData.EntityName);
-                    }
-                }
+                foundOne = true;
+                if (stationNames.Length > 0)
+                        stationNames.Append('\n');
+
+                stationNames.Append(meta.EntityName);
```

Comment:
This could probably be re-written as a filter and then a string join. I guess you're not trying to fix this now?